doc/manual/src/architecture/file-system-object.md

```diff
@@ -0,0 +1,63 @@
+# File System Object
+
+Nix uses a simplified model of the file system, which consists of file system objects.
+Every file system object is one of the following:
+
+ - File
+
+   - A possibly empty sequence of bytes for contents
+   - A single boolean representing the [executable](https://en.m.wikipedia.org/wiki/File-system_permissions#Permissions) permission
+
+ - Directory
+
+   Mapping of names to child file system objects
+
+ - [Symbolic link](https://en.m.wikipedia.org/wiki/Symbolic_link)
+
+   A file system path that may point anywhere
```

Comment:
```suggestion
   An arbitrary string
```
or something like that. We don't actually assign any semantics to symlinks.

Reply:
```suggestion
   An arbitrary string.
   Nix does not assign any semantics to symbolic links.
```

Reply:
```suggestion
   An arbitrary string.
   Nix does not assign any semantics to the target of symbolic links.
```